src/com/owncloud/android/ui/activity/FileActivity.java

```diff
@@ -978,26 +524,24 @@ public void onServiceDisconnected(ComponentName component) {
         }
     }
 
-
     @Override
     public FileDownloaderBinder getFileDownloaderBinder() {
         return mDownloaderBinder;
     }
 
-
     @Override
     public FileUploaderBinder getFileUploaderBinder() {
         return mUploaderBinder;
     }
 
-
+    @Override
     public void restart(){
         Intent i = new Intent(this, FileDisplayActivity.class);
         i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-        i.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
```

Comment:
OK, let's forget this. That flag is correctly removed, otherwise account switching doesn't work. I can't find the weird navigation behavior I remembered.